### PR TITLE
chore(react): initial scaffold

### PR DESCRIPTION
**Pull request by alexcasche**
**What This PR Does**
- sets up a vite scaffold for react package - modeled after vue package

**Next Steps**
- decide what Providers/Composables should exist
- decide where shared code should live (utils/types)

### CODE REVIEW

**Review by mdarrik — COMMENTED** [review]
So one other question, since React hooks/projects have good TS support, do we want to write this in TS instead of vanilla JS so the teams that are using TS can take advantage of it?

**Comment by alexcasche** [comment]
> So one other question, since React hooks/projects have good TS support, do we want to write this in TS instead of vanilla JS so the teams that are using TS can take advantage of it?

@mdarrik  Yup the intention is to make this Typescript.  I'm wondering where we want these types to live.

File: packages/react/README.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+[![npm version](https://img.shields.io/npm/v/@nacelle/react?style=for-the-badge)](https://npmjs.org/package/@nacelle/react)
+
+## Installation
+
+```
+npm i @nacelle/react
+```
+
+`@nacelle/react` also depends on [`@nacelle/client-js-sdk`](https://www.npmjs.com/package/@nacelle/client-js-sdk). Follow those package's installation instructions before using `@nacelle/react` in your React.js + Nacelle-powered storefront.
```
[the remark this answers is not in the document]
Comment (alexcasche) [reply packages/react/README.md:13]: The providers won't need access to the sdk.  But the `SpaceProvider` exposes a global instance so settings can be shared app-wide.  At least that was the intention in Vue.

**Review by KrisQ — APPROVED** [review]
Looks good!

**Comment by NWRichmond** [comment]
This scaffold looks great. But I'm going to close this for now until we have a chance to revisit the project goals. We can reopen the PR later as needed.

**Pull request closed without merging**